wagtail/tests/test_sites.py

```diff
@@ -216,3 +217,140 @@ def test_result_order_when_multiple_sites_share_the_same_root_page(self):
         # Followed by entries for others in 'host' alphabetical order
         self.assertEqual(result[1][0], self.abc_site.id)
         self.assertEqual(result[2][0], self.def_site.id)
+
+
+class TestSiteCache(TestCase):
+    fixtures = ["test.json"]
+
+    def setUp(self):
+        super().setUp()
+        # Tests run in a shared thread, and loading of fixture data does not
+        # trigger clearing of threadlocal caches. So, we manually clear them
+        # here, so that cached site/root path values always reflect the data
+        # in the fixture
+        Site.clear_caches()
+
+    def test_cache(self):
+        # The cache should be empty to start with
+        self.assertIsNone(Site.objects._get_cached_list())
+
+        # Requesting the list should result in a database query
+        with self.assertNumQueries(1):
+            value = Site.objects.get_cached_list()
+
+        # Check return value matches expectations
+        self.assertEqual(value, tuple(Site.objects.get_site_root_path_ordered()))
+
+        # Requesting again should return the cached value
+        with self.assertNumQueries(0):
+            second_value = Site.objects.get_cached_list()
+
+        # The returned tuples should be equal, but should not be the same tuple object
+        self.assertEqual(value, second_value)
+        self.assertIsNot(value, second_value)
+
+    def test_cache_clears_when_site_saved(self):
+        """
+        This tests that the cache is cleared whenever a site is saved
+        """
+        # Trigger cache population
+        Site.objects.get_cached_list()
+
+        # Save a site
+        Site.objects.get(is_default_site=True).save()
+
+        # Check the cache was cleared
+        self.assertIsNone(Site.objects._get_cached_list())
+
+    def test_cache_clears_when_site_deleted(self):
+        """
+        This tests that the cache is cleared whenever a site is deleted
+        """
+        # Trigger cache population
+        Site.objects.get_cached_list()
+
+        # Delete a site
+        Site.objects.get(is_default_site=True).delete()
+
+        # Check the cache was cleared
+        self.assertIsNone(Site.objects._get_cached_list())
+
+    def test_cache_clears_when_site_root_moves(self):
+        """
+        This tests for an issue where if a site root page was moved, all
+        the page urls in that site would change to None.
```

Comment:
Question: This test doesn't seem to assert the URL

Reply:
Sorry, yes. Copied from the SiteRootPath tests without revising the docstring, but the steps are slightly different.

Reply:
These test were removed in f51287a in favour of extending TestSiteRootPathsCache to cover clearing of both caches